src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -209,20 +209,53 @@ static unsigned uct_cuda_copy_iface_progress(uct_iface_h tl_iface)
     return count;
 }
 
+static ucs_status_t uct_cuda_copy_iface_event_fd_get(uct_iface_h tl_iface, int *fd_p)
+{
+    uct_cuda_copy_iface_t *iface = ucs_derived_of(tl_iface, uct_cuda_copy_iface_t);
+
+    if (-1 == iface->eventfd) {
+        iface->eventfd = eventfd(0, EFD_CLOEXEC | EFD_NONBLOCK);
+        if (iface->eventfd == -1) {
+            ucs_error("Failed to create event fd: %m");
+            return UCS_ERR_IO_ERROR;
+        }
+    }
+
+    *fd_p = iface->eventfd;
+    return UCS_OK;
+}
+
+static void uct_cuda_copy_common_cb(void *cuda_copy_iface)
+{
+    uct_cuda_copy_iface_t *iface = cuda_copy_iface;
+    uint64_t dummy = 1;
+    int ret;
+
+    do {
+        ret = write(iface->eventfd, &dummy, sizeof(dummy));
+        if (ret == sizeof(dummy)) {
+            return;
+        } else if (ret == -1) {
+            if (errno == EAGAIN) {
+                continue;
+            } else if (errno != EINTR) {
+                ucs_error("Signaling wakeup failed: %m");
+                return;
+            }
+        } else {
+            ucs_assert(ret == 0);
+        }
+    } while (ret == 0);
```

Comment:
as discussed offline need to move this logic to ucs to avoid copy/paste with ucp code. also, could reuse code for read/write.